spec/factories/courses.rb

```diff
@@ -255,5 +259,20 @@
     trait :closed do
       application_status { :closed }
     end
+
+    trait :published_teacher_degree_apprenticeship do
```

Comment:
After giving so much thought, I consciously didn't choose `:published_undergraduate`. Let me know if you have strong opinions to be undergraduate here.